### PR TITLE
fix for discriminator gradients setup in generator

### PR DESCRIPTION
**Pull request by czero69**
Hello,

I have recently begun reviewing the code and noticed something that might be important for the training process. During the generator's training phase, should the gradients for the discriminator be disabled? This adjustment could potentially lead to improved training results.

Best regards,
Kamil

### CODE REVIEW

**Comment by BonifazStuhr** [comment]
Hello Kamil,

thanks for checking the code! Setting the flag to false should be the more efficient/clean way, so I'm happy to merge your request. However, the optimizer of the generator does not have access to the parameters of the discriminator:

From https://github.com/BonifazStuhr/feamgan/blob/main/feamgan/Experiment_Component/Models/FeaMGAN.py: 

```
G_params = list(self.netG.parameters())
self.optimizer_G = torch.optim.AdamW(G_params, lr=G_lr, betas=(beta1, beta2), weight_decay=0.0001) 
```

We also do not perform a step with the optimizer of the discriminator in the `generator` function. Therefore, in my understanding, this should not affect the training of the discriminator. 

However, setting the flag to false is the clearer version!

Yours sincerely,
Bonifaz Stuhr
